data/transactions/transaction.go

```diff
@@ -143,43 +136,22 @@ type TxGroup struct {
 	// together, sequentially, in a block in order for the group to be
 	// valid.  Each hash in the list is a hash of a transaction with
 	// the `Group` field omitted.
-	TxGroupHashes []crypto.Digest `codec:"txlist"`
+	TxGroupHashes []crypto.Digest `codec:"txlist,allocbound=config.Consensus[protocol.ConsensusCurrentVersion].MaxTxGroupSize"`
```

Comment:
I wonder how error-prone this will be in event of a protocol upgrade.

Reply:
This is indeed one of the more subtle bounds.  I think this works out OK as long as we are increasing the limit upward.  If we decrease the limit at some point, I think we should keep the allocation bound high for compatibility, for at least one round of upgrades.

Reply:
It does make things more awkward in the cases where we're having multiple networks (betanet vs. mainnet/testnet) updating through different versions prior to a software release. I guess there we'd manually set it to the maximum of all those values.

Reply:
Thinking about it more, this allocbound is one of the ones that should have been `allocbound=-` because we never decode these structs -- we only encode them to compute the txgroup hash..